src/main/java/org/ibp/api/java/impl/middleware/common/validator/AttributeValidator.java

```diff
@@ -27,11 +48,111 @@ public void validateAttributeIds(final BindingResult errors, final List<String>
 		final List<Attribute> attributes = this.germplasmDataManager.getAttributeByIds(ids);
 		if (attributes.size() != ids.size()) {
 			errors.reject("attribute.invalid", "");
+		}
+	}
+
+	public void validateAttributeType(final BindingResult errors, final String attributeType) {
+		if (StringUtils.isEmpty(attributeType)) {
 			return;
 		}
+
+		if (!AttributeValidator.ALLOWED_ATTRIBUTE_TYPES.contains(attributeType.toUpperCase())) {
+			errors.reject("attribute.type.invalid", "");
+			throw new ApiRequestValidationException(errors.getAllErrors());
+		}
+	}
+
+	void validateAttributeCode(final BindingResult errors, final GermplasmAttributeRequestDto dto) {
+		final List<AttributeDTO> attributeDTOS = this.germplasmAttributeService
+			.filterGermplasmAttributes(Collections.singleton(dto.getAttributeCode()), dto.getAttributeType());
```

Comment:
I havent tried it but does it work when passing the type in lowercase?

Reply:
Hi Clary, I tried and it works regardless of the case of the letters.